util/bulk_process_tx.py

```diff
@@ -161,6 +165,23 @@ def process_job(self):
             # change the jobs array to None (from an empty array) to indicate we are done and can shut down
             self.jobs = None
 
+    async def wallet_sync(self, start_block=None, rebuild=False):
+        if self.wallet_needs_recovery:
+            self.wallet_needs_recovery = False
```

Comment:
So I see you are flipping this before you start the wallet recovery, presumably so that if another thread gets in here it will not start another recovery... Do we need another flag to indicate that you are already recovering to prevent the else clause from starting a sync while a recovery is in process?

Reply:
Well, python isn't multi-threaded like java. It's a single thread with asynchronous execution like Node.js. The concern here isn't that another thread/process might call this a second time. The issue is that `recover_wallet` can call `wallet_sync` and unless you already flipped this flag, you'll get into an infinite loop. So, we turn the flag off before doing the recovery so that subsequent calls to `wallet_sync` will not try to perpetually recover.

The way this code is structured, there won't be any concurrent calls to this method, so that's not a concern. Thus, no additional flag is needed...

Reply:
Okay, cool. That makes sense, and it's good to know that our Python process is single threaded.